src/Orleans.Core.Abstractions/Core/Grain.cs

```diff
@@ -185,7 +185,9 @@ protected virtual IStreamProvider GetStreamProvider(string name)
                 throw new ArgumentNullException(nameof(name));
                 
             EnsureRuntime();
-            return this.ServiceProvider.GetServiceByName<IStreamProvider>(name);
+            IStreamProvider streamProvider = this.ServiceProvider.GetServiceByName<IStreamProvider>(name);
+            if (streamProvider == null) throw new KeyNotFoundException(name);
```

Comment:
I think we need to throw a more descriptive exception message here. Before the message is something like "stream provider with name blablba is not configured". If we suddenly change the exception message and also changed the whole provider infra. people may get confused.